services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTestImpl.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+
+import java.nio.file.Path
+import scala.concurrent.Future
+
+class ProjectFileServiceTestImpl(projectFiles: Seq[ProjectFile], testMode: TestMode) extends ProjectFileService {
+
+  override def validateFile(fileContent: ProjectFileContent): Future[Either[ValidationError, Unit]] =
+    testMode match {
```

Comment:
I don't really get the point of TestMode. If you could explain it to me it would be great. Everything else seems right to me

Reply:
It is a marker, which tells testImpl to return a result needed for current testcase. It allows to simulate two types of behavior - success or failure inside one class. 
`val projectFileService = ProjectFileServiceTestImpl(dummyFile)` - if we want a method to be successfull or
`val projectFileService = ProjectFileServiceTestImpl.withException(someException)` - if we want a method with an exception.
In both cases these values are instances of the same `ProjectFileServiceTestImpl`. `TestMode` is needed to tell the test implementation which type of behavior it must choose.

Other way could be creating two different test Implementations, for example `ProjectFileServiceFailure` - it's overridden methods always fail, and `ProjectFileServiceSuccessfull` - always works without errors.